### ISSUE TITLE
[Accessiblity] Narrator did not announce anchor property check state when press space to check

### ISSUE THREAD

**Issue by Liv-Goh** [dotnet/winforms#12515]
### .NET version

.NET SDK 10.0.100-alpha.1.24562.4

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, this is not a regression issue. This issue repro in previous versions of .NET too

## Current Behavior
In runtime property grid, Narrator only announces "space" when checked using spacebar

https://github.com/user-attachments/assets/4d6ea558-e4b0-4118-a835-358661ee4813

## Expected Behavior
In VS property grid, Narrator announces "space, Top/Right/Left/Bottom checkbox checked"

https://github.com/user-attachments/assets/fbfabe8c-710f-497f-8cd0-e5923220fdd9

### Steps to reproduce

1. Create a .NET Core Winforms project.
2. Add a button and PropertyGrid controls in form designer.
3. Set the button as propertyGrid SelectedObject
4. Build and run project.
5. Open Narrator tool and select Anchor property in propertyGrid control.
6. Use keyboard spacebar to change the status of the anchor direction to checked and unchecked

## More Info
This issue repro in NVDA tool.

**Comment by Tanya-Solyanik** [comment]
@merriemcgaw  - this is "by design" for the custom drop down type editors.  To select one of the predefined values with keyboard, the user should press SPACE, followed by Down or Up arrows, while the drop down is not expanded. And the user can only select a single Anchor value this way, to select both Left and Right, the user should type in the text "Left, Right" and then commit it with pressing ENTER. Keyboard experience is not blocked, but the user has to know the value to type in. I'm not sure how to fix this though. As is the source code editor experience is not convenient that that in the property grid...

**Comment by merriemcgaw** [comment]
This is by design.
